src/xrpld/ledger/detail/ApplyStateTable.cpp

```diff
@@ -147,7 +159,8 @@ ApplyStateTable::apply(
             }
             auto const origNode = to.read(keylet::unchecked(item.first));
             auto curNode = item.second.second;
-            if ((type == &sfModifiedNode) && (*curNode == *origNode))
+            if ((type == &sfModifiedNode) && (*curNode == *origNode) &&
+                !isBatch)
```

Comment:
if i'm understanding correctly, when the transaction type is `Batch`, we are going to  add the previous and final fields of the object even though they are the same. 

This doesn't sound too correct, please correct me if i'm wrong. TY

Reply:
We are going to add the `batchPrevAcctRootFields`. The curNode and origNode are the same on the batch txn because of the way its reapplied in `updateAccountRootEntry`. I guess handling this as its own case might be preferred for better understanding?

Reply:
we can just add a comment here to explain the situation. Also it might be worth add an assertion to make sure that nodes are not modified if it is a batch transaction - this might before useful for future debugging

```
bool const nodesUnmodified = *curNode == *origNode;
if ((type == &sfModifiedNode) && nodesUnmodified &&
                !isBatch)
     continue;

assert(isBatch && !nodesUnmodified)
```